main.js

```diff
@@ -54,3 +55,29 @@ function dtsPath(path) {
 	const { dir, name, ext } = parsePath(path);
 	return join(dir, `${name}.d${ext}.ts`);
 }
+
+// The same camelCase function from css-loader that is used to provide
+// the class selector names to Gatsby.
+function camelCase(input) {
+  let result = input.trim();
+
+  if (result.length === 0) {
+    return "";
+  }
+
+  if (result.length === 1) {
+    return result.toLowerCase();
+  }
+
+  const hasUpperCase = result !== result.toLowerCase();
+
+  if (hasUpperCase) {
+    result = preserveCamelCase(result);
+  }
+
+  return result
+    .replace(/^[_.\- ]+/, "")
+    .toLowerCase()
+    .replace(/[_.\- ]+([\p{Alpha}\p{N}_]|$)/gu, (_, p1) => p1.toUpperCase())
+    .replace(/\d+([\p{Alpha}\p{N}_]|$)/gu, (m) => m.toUpperCase());
```

Comment:
Can you name the regex values (local variables) or add comments for readability? I think it says

1. Remove leading invalid chars (`_`, `.`, `-`, and ` ` (space)). _Unsure why only these though._
2. Lower-case the entire string
3. Replace invalid chars (same as (1)) and the next character with the upper case of that next character. _Assuming the `_` is the full match and `p1` is the captured group._
4. Replace numbers followed by a character with the upper case of the full match. _I guess this exists to “(re)normalize” the string following the lower-casing of (2)?_